app/uk/gov/hmrc/serviceconfigs/service/SlugConfigUpdateHandler.scala

```diff
@@ -82,23 +82,29 @@ class SlugConfigUpdateHandler  @Inject()
     (eitherSlugInfo match {
       case Left(error) => Future(Left(error))
       case Right(slugMessage) =>
-        slugConfigurationService.addSlugInfo(slugMessage.info)
-          .map(saveResult => if (saveResult) Right(()) else Left(s"SlugInfo for message (ID '${message.messageId()}') was sent on but not saved."))
-          .recover {
-            case e =>
-              val errorMessage = s"Could not store slug info for message with ID '${message.messageId()}'"
-              Logger.error(errorMessage, e)
-              Left(s"$errorMessage ${e.getMessage}")
-          }
-
-        slugConfigurationService.addDependencyConfigurations(slugMessage.configs)
-          .map(saveResult => if (saveResult.forall(_ == true)) Right(()) else Left(s"Configuration for message (ID '${message.messageId()}') was sent on but not saved."))
-          .recover {
-            case e =>
-              val errorMessage = s"Could not store slug configuration for message with ID '${message.messageId()}'"
-              Logger.error(errorMessage, e)
-              Left(s"$errorMessage ${e.getMessage}")
-          }
+        for {
+          si <- slugConfigurationService.addSlugInfo(slugMessage.info)
+            .map(saveResult => if (saveResult) Right(()) else Left(s"SlugInfo for message (ID '${message.messageId()}') was sent on but not saved."))
+            .recover {
+              case e =>
+                val errorMessage = s"Could not store slug info for message with ID '${message.messageId()}'"
+                Logger.error(errorMessage, e)
+                Left(s"$errorMessage ${e.getMessage}")
+            }
+          dc <- slugConfigurationService.addDependencyConfigurations(slugMessage.configs)
+            .map(saveResult => if (saveResult.forall(_ == true)) Right(()) else Left(s"Configuration for message (ID '${message.messageId()}') was sent on but not saved."))
+            .recover {
+              case e =>
+                val errorMessage = s"Could not store slug configuration for message with ID '${message.messageId()}'"
+                Logger.error(errorMessage, e)
+                Left(s"$errorMessage ${e.getMessage}")
+            }
```

Comment:
Why don't we provide an addSlugMessage method in SlugConfigurationService, to abstract away the different error logging for failure to store slug data or config?